staging/src/k8s.io/component-helpers/storage/volume/pv_helpers.go

```diff
@@ -279,6 +282,10 @@ func FindMatchingVolume(
 		if GetPersistentVolumeClass(volume) != requestedClass {
 			continue
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) &&
+			ptr.Deref(volume.Spec.VolumeAttributesClassName, "") != ptr.Deref(claim.Spec.VolumeAttributesClassName, "") {
+			continue
+		}
```

Comment:
Hi @sunnylovestiramisu @msau42 

If pvc's vac is nil, can it matches with such pvc which vac is empty string? If pvc's vac is an empty string, can it matches with such pvc which vac is nil?

Reply:
And should feature-gate be removed because it import k/k in staging?

Reply:
If pvc's vac is nil, can it matches with such pv which vac is empty string? If pvc's vac is an empty string, can it matches with such pv which vac is nil?
<-- We discussed that we do not allow passing in empty string for VAC name? But if it somehow end up with an empty string, it means that the pv/pvc does not have a vac, this makes empty string kinda equal to nil. I think we can match it. But should we reset the empty string back to nil?

Reply:
should feature-gate be removed because it import k/k in staging?

I am not sure if there is any harm of keeping the feature gate, will we ever change to not import k/k staging? If there is possibility, keeping the feature gate is like a safe guard.

Reply:
> But should we reset the empty string back to nil?

It is disallowed. https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L2398

Reply:
> Can we add unit tests for these two cases you mentioned as well?

If the pr's changes are reviewed and accepted, I will add some unit tests.